content/2020-06-19-how-to-learn-typescript-quickly/index.md

```diff
@@ -0,0 +1,259 @@
+
+With this vision in mind, you now have clarity on it means when you say "learn TypeScript." and you know what outcomes you expect to achieve.
+
+### Establish a Timeframe
+
+You don't want to be racing on this map forever. You need to be realistic and ask yourself how much time we have for this. Estimating time to complete tasks in software is hard. We'll try our best by using something I like to call the **Timeframe Formula**:
+
+Take the length of months you want to work on this, multiple by 4 to give the amount in weeks. Then multiple it by the number of hours per week that you want to dedicate.
+
+https://excalidraw.com/#json=4877101601652736,SeRw5x2aEzwtztaHHk0xUQ
+
+For example, when I did a similar project, I gave myself 3 months with 8 hours per week. Using the Timeframe Formula, you get:
+
+(3 months * 4 weeks/month) * 8 hours/week = 96 hours total.
+
+https://excalidraw.com/#json=5134809571524608,ASGbR83wxxoafOuInmZ5Ew
+
+You now have your timeframe and expectation of how much you plan to dedicate each week.
+
+### Build a Timeline
+
+The next thing we need to add to our map is a timeline. Think of this as a week by week calendar. It helps you think less because you have already decided what to do for that week. It won't be perfect. You may need to readjust, but at least you will have established the framing.
+
+An example:
+
+- [ ] Week 1 (June 1)
+  - [ ] Programming TypeScript
+    - [ ] Chapter 1 - Introduction
+    - [ ] Chapter 2 - TypeScript: a 10_000 foot view
+      - [ ] Book exercises
+    - [ ] Chapter 3 - All About Types
+      - [ ] Book exercises
+  - [ ] Official TypeScript Handbook (OTH)
+    - [ ] Basic Types
+    - [ ] Variable Declarations
+
+You know yourself best. Think back to the Timeframe Formula. The example estimated 8hrs/week and 12 weeks in total. Knowing the speed that you read it, you would adjust accordingly. A complete timeline would have all the weeks (i.e. 12 in this case) filled out.
+
+Two important tips:
+
+1. Leave 1-2 weeks buffer towards the end
+2. Give yourself some breathing room
+
+Life comes up and you may fall behind. That's where the buffer comes in! Some weeks, you may find that you complete everything in 6 hours. No need to fill those two other hours if you don't want to. Reward yourself with rest.
+
+With your timeline, you'll be good to go!
+
+### Keep it Organized
+
+You want to stay organized on this journey. That means a place to keep track of your timeline, log any notes or things you learn, and have visual indicators for the progress you're making.
+
+I use Notion for this because it supports this type of project management well. I make a place for the project. I add the timeline and use checkboxes. Then, I check them off as I go and take notes as I learn new things.
```

Comment:
```suggestion
I use Notion for this because it supports project management well. I make a place for the project. I add the timeline and use checkboxes. Then, I check them off as I go and take notes as I'm learning.
```